iceoryx2/tests/service_request_response_builder_tests.rs

```diff
@@ -0,0 +1,956 @@
+    }
+
+    #[test]
+    fn opened_service_reads_config_correctly<Sut: Service>() {
+        let service_name = generate_service_name();
+        let config = generate_isolated_config();
+        let node = NodeBuilder::new().config(&config).create::<Sut>().unwrap();
+
+        let _sut_create = node
+            .service_builder(&service_name)
+            .request_response::<u64, u64>()
+            .enable_safe_overflow_for_requests(false)
+            .enable_safe_overflow_for_responses(false)
+            .max_active_requests(1)
+            .max_active_responses(2)
+            .max_borrowed_requests(3)
+            .max_borrowed_responses(4)
+            .max_request_buffer_size(5)
+            .max_response_buffer_size(6)
+            .max_servers(7)
+            .max_clients(8)
+            .max_nodes(9)
+            .create()
+            .unwrap();
+
+        let sut_open = node
+            .service_builder(&service_name)
+            .request_response::<u64, u64>()
+            .open()
+            .unwrap();
+
+        assert_that!(sut_open.static_config().has_safe_overflow_for_requests(), eq false);
+        assert_that!(sut_open.static_config().has_safe_overflow_for_responses(), eq false);
+        assert_that!(sut_open.static_config().max_active_requests(), eq 1);
+        assert_that!(sut_open.static_config().max_active_responses(), eq 2);
+        assert_that!(sut_open.static_config().max_borrowed_requests(), eq 3);
+        assert_that!(sut_open.static_config().max_borrowed_responses(), eq 4);
+        assert_that!(sut_open.static_config().max_request_buffer_size(), eq 5);
+        assert_that!(sut_open.static_config().max_response_buffer_size(), eq 6);
+        assert_that!(sut_open.static_config().max_servers(), eq 7);
+        assert_that!(sut_open.static_config().max_clients(), eq 8);
+        assert_that!(sut_open.static_config().max_nodes(), eq 9);
+    }
+
+    #[test]
+    fn list_finds_created_services<Sut: Service>() {
+        const NUMBER_OF_SERVICES: usize = 12;
+        let config = generate_isolated_config();
+        let node = NodeBuilder::new().config(&config).create::<Sut>().unwrap();
+        let mut services = vec![];
+        let mut service_names = vec![];
+
+        for _ in 0..NUMBER_OF_SERVICES {
+            let service_name = generate_service_name();
+            let sut = node
+                .service_builder(&service_name)
+                .request_response::<u64, u64>()
+                .create()
+                .unwrap();
+            services.push(sut);
+            service_names.push(service_name);
+        }
+
+        for name in &service_names {
+            assert_that!(Sut::does_exist(name, &config, MessagingPattern::RequestResponse).unwrap(), eq true);
+        }
+
+        let mut number_of_listed_services = 0;
+        Sut::list(&config, |service| {
+            assert_that!(service_names, contains * service.static_details.name());
```

Comment:
Food for thought. Shouldn't the name be removed from the `service_names` in order to detect getting the same name all the time?